Kovri doesn't build with Boost 1.67

---
**By submitting this issue, I confirm the following:**

- I have read and understood the contributor guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.
---

Compilers throw the following errors:

MSVC
```
kovri\src\client\address_book\impl.cc(92): error C2440: '<function-style-cast>': cannot convert from 'kovri::client::AddressBookDefaults::SubscriberTimeout' to 'boost::posix_time::minutes'
kovri\src\client\address_book\impl.cc(94): note: No constructor could take the source type, or constructor overload resolution was ambiguous
```

clang
```
kovri\src\client\address_book\impl.cc(92): error: no matching conversion for functional-style cast from 'kovri::client::AddressBookDefaults' to 'boost::posix_time::minutes'
```

gcc
```
kovri\src\client\address_book\impl.cc(92): error: no matching function for call to 'boost::posix_time::minutes::minutes(kovri::client::AddressBookDefaults::SubscriberTimeout)'
```

@coneiric The failure of [buildbot/kovri-all-ubuntu-amd64](https://build.getmonero.org/builders/kovri-all-ubuntu-amd64/builds/956) isn't caused by this changes or boost.
```
Assembler messages:
Fatal error: can't create CMakeFiles/kovri-util.dir/cpuid.cc.o: No such file or directory
src/util/CMakeFiles/kovri-util.dir/build.make:254: recipe for target 'src/util/CMakeFiles/kovri-util.dir/cpuid.cc.o' failed
```

Something is wrong with that buildbot.
Other builds which use boost 1.58 are green - [buildbot/kovri-all-ubuntu-i686](https://build.getmonero.org/builders/kovri-all-ubuntu-i686/builds/899)

--- a/src/client/address_book/impl.cc
+++ b/src/client/address_book/impl.cc
@@ -89,7 +89,7 @@ void AddressBook::Start(
     std::make_unique<boost::asio::deadline_timer>(
         m_SharedLocalDestination->GetService());
   m_SubscriberUpdateTimer->expires_from_now(
-      boost::posix_time::minutes(SubscriberTimeout::InitialUpdate));
+      boost::posix_time::minutes{static_cast<long>(SubscriberTimeout::InitialUpdate)});
   m_SubscriberUpdateTimer->async_wait(
       std::bind(
           &AddressBook::SubscriberUpdateTimer,
@@ -120,7 +120,7 @@ void AddressBook::SubscriberUpdateTimer(
     }
     // Try again after timeout
     m_SubscriberUpdateTimer->expires_from_now(
-        boost::posix_time::minutes(SubscriberTimeout::InitialRetry));
+        boost::posix_time::minutes{static_cast<long>(SubscriberTimeout::InitialRetry)});
     m_SubscriberUpdateTimer->async_wait(
         std::bind(
             &AddressBook::SubscriberUpdateTimer,
@@ -272,10 +272,11 @@ void AddressBook::HostsDownloadComplete(
   LOG(debug) << "AddressBook: subscription download complete";
   if (m_SubscriberUpdateTimer) {
     m_SubscriberUpdateTimer->expires_from_now(
-        boost::posix_time::minutes(
+        boost::posix_time::minutes{
+        static_cast<long>(
             success
             ? SubscriberTimeout::ContinuousUpdate
-            : SubscriberTimeout::ContinuousRetry));
+            : SubscriberTimeout::ContinuousRetry)});
     m_SubscriberUpdateTimer->async_wait(
         std::bind(
             &AddressBook::SubscriberUpdateTimer,

--- a/src/core/router/transports/ssu/data.cc
+++ b/src/core/router/transports/ssu/data.cc
@@ -509,8 +509,8 @@ void SSUData::ScheduleResend() {
     << "scheduling resend";
   m_ResendTimer.cancel();
   m_ResendTimer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::ResendInterval));
+      boost::posix_time::seconds{
+        static_cast<long>(SSUDuration::ResendInterval)});
   auto s = m_Session.shared_from_this();
   m_ResendTimer.async_wait(
       [s](const boost::system::error_code& ecode) {
@@ -562,8 +562,8 @@ void SSUData::ScheduleDecay() {
     << "SSUData:" << m_Session.GetFormattedSessionInfo() << "scheduling decay";
   m_DecayTimer.cancel();
   m_DecayTimer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::DecayInterval));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::DecayInterval)});
   auto s = m_Session.shared_from_this();
   m_ResendTimer.async_wait(
       [s](const boost::system::error_code& ecode) {
@@ -586,8 +586,8 @@ void SSUData::ScheduleIncompleteMessagesCleanup() {
     << "scheduling incomplete messages cleanup";
   m_IncompleteMessagesCleanupTimer.cancel();
   m_IncompleteMessagesCleanupTimer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::IncompleteMessagesCleanupTimeout));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::IncompleteMessagesCleanupTimeout)});
   auto s = m_Session.shared_from_this();
   m_IncompleteMessagesCleanupTimer.async_wait(
       [s](const boost::system::error_code& ecode) {

--- a/src/core/router/transports/ssu/server.cc
+++ b/src/core/router/transports/ssu/server.cc
@@ -476,8 +476,8 @@ std::set<SSUSession *> SSUServer::FindIntroducers(
 void SSUServer::ScheduleIntroducersUpdateTimer() {
   LOG(debug) << "SSUServer: scheduling introducers update timer";
   m_IntroducersUpdateTimer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::KeepAliveInterval));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::KeepAliveInterval)});
   m_IntroducersUpdateTimer.async_wait(
       std::bind(
           &SSUServer::HandleIntroducersUpdateTimer,
@@ -590,8 +590,8 @@ void SSUServer::RemovePeerTest(
 void SSUServer::SchedulePeerTestsCleanupTimer() {
   LOG(debug) << "SSUServer: scheduling PeerTests cleanup timer";
   m_PeerTestsCleanupTimer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::PeerTestTimeout));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::PeerTestTimeout)});
   m_PeerTestsCleanupTimer.async_wait(
       std::bind(
           &SSUServer::HandlePeerTestsCleanupTimer,

--- a/src/core/router/transports/ssu/session.cc
+++ b/src/core/router/transports/ssu/session.cc
@@ -1582,8 +1582,8 @@ void SSUSession::WaitForConnect() {
 void SSUSession::ScheduleConnectTimer() {
   m_Timer.cancel();  // TODO(anonimal): cancel is called within expires_from_now
   m_Timer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::ConnectTimeout));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::ConnectTimeout)});
   m_Timer.async_wait(
       std::bind(
           &SSUSession::HandleConnectTimer,
@@ -1609,8 +1609,8 @@ void SSUSession::Introduce(
   if (m_State == SessionState::Unknown) {
     // set connect timer
     m_Timer.expires_from_now(
-        boost::posix_time::seconds(
-            SSUDuration::ConnectTimeout));
+        boost::posix_time::seconds{
+            static_cast<long>(SSUDuration::ConnectTimeout)});
     m_Timer.async_wait(
         std::bind(
           &SSUSession::HandleConnectTimer,
@@ -1624,8 +1624,8 @@ void SSUSession::WaitForIntroduction() {
   m_State = SessionState::Introduced;
   // set connect timer
   m_Timer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::ConnectTimeout));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::ConnectTimeout)});
   m_Timer.async_wait(
       std::bind(
         &SSUSession::HandleConnectTimer,
@@ -1678,8 +1678,8 @@ void SSUSession::Failed() {
 void SSUSession::ScheduleTermination() {
   m_Timer.cancel();
   m_Timer.expires_from_now(
-      boost::posix_time::seconds(
-          SSUDuration::TerminationTimeout));
+      boost::posix_time::seconds{
+          static_cast<long>(SSUDuration::TerminationTimeout)});
   m_Timer.async_wait(
       std::bind(
           &SSUSession::HandleTerminationTimer,

